### ISSUE TITLE
Can't use MinGW to make freeserf (HEAD).

### ISSUE THREAD

**Issue by ryan-de-boer** [freeserf/freeserf#378]
I've got it working up to getting 'make' to work. It keeps saying no such file or directory.
`
e:\Dev\freeserf\build>cmake -G "MinGW Makefiles" -D ENABLE_XMP:BOOL=OFF -D SDL2_
INCLUDE_DIR="E:\Dev\freeserf\dependencies\SDL2-2.0.7\i686-w64-mingw32\include" -
D SDL2_LIBRARY="E:\Dev\freeserf\dependencies\SDL2-2.0.7\i686-w64-mingw32\lib\lib
SDL2.a" -D SDL2_MAIN_LIBRARY="E:\Dev\freeserf\dependencies\SDL2-2.0.7\i686-w64-m
ingw32\lib\libSDL2main.a" -D SDL2_MIXER_INCLUDE_DIR="E:\Dev\freeserf\dependencie
s\SDL2_mixer-2.0.2\i686-w64-mingw32\include" -D SDL2_MIXER_LIBRARY="E:\Dev\frees
erf\dependencies\SDL2_mixer-2.0.2\i686-w64-mingw32\lib\libSDL2_mixer.a" -D SDL2_
IMAGE_INCLUDE_DIR="E:\Dev\freeserf\dependencies\SDL2_image-2.0.2\i686-w64-mingw3
2\include" -D SDL2_IMAGE_LIBRARY="E:\Dev\freeserf\dependencies\SDL2_image-2.0.2\
i686-w64-mingw32\lib\libSDL2_image.a" ..
-- FREESERF_VERSION = v0.2-rc1-354-g3e88595
-- PROJECT_VERSION = 0.2.3e88595
-- PROJECT_VERSION_MAJOR = 0
-- PROJECT_VERSION_MINOR = 2
-- PROJECT_VERSION_PATCH = 3e88595
-- Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE)
-- Could NOT find XMP (missing: XMP_LIBRARY XMP_INCLUDE_DIR)
-- Configuring done
-- Generating done
-- Build files have been written to: E:/Dev/freeserf/build
`

`
e:\Dev\freeserf\build>make
[  1%] Building CXX object src/CMakeFiles/game.dir/building.cc.obj
G__~1.EXE: error: /wd4290: No such file or directory
make[2]: *** [src\CMakeFiles\game.dir\build.make:63: src/CMakeFiles/game.dir/bui
lding.cc.obj] Error 1
make[1]: *** [CMakeFiles\Makefile2:87: src/CMakeFiles/game.dir/all] Error 2
make: *** [Makefile:162: all] Error 2
`

Here is what build.make has:

`
src/CMakeFiles/game.dir/building.cc.obj: src/CMakeFiles/game.dir/flags.make
src/CMakeFiles/game.dir/building.cc.obj: src/CMakeFiles/game.dir/includes_CXX.rsp
src/CMakeFiles/game.dir/building.cc.obj: ../src/building.cc
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --green --progress-dir=E:\Dev\freeserf\build\CMakeFiles --progress-num=$(CMAKE_PROGRESS_1) "Building CXX object src/CMakeFiles/game.dir/building.cc.obj"
	cd /d E:\Dev\freeserf\build\src && E:\PROGRA~1\MINGW-~1\I686-7~1.0-P\mingw32\bin\G__~1.EXE  $(CXX_DEFINES) $(CXX_INCLUDES) $(CXX_FLAGS) -o CMakeFiles\game.dir\building.cc.obj -c E:\Dev\freeserf\src\building.cc
`

Its a shame it can't just find ../src/building.cc

I've attached a txt file that might be easier to read (formatting gets lost in the post)
[Can't use MinGW post.txt](https://github.com/freeserf/freeserf/files/1628447/Can.t.use.MinGW.post.txt)

**Comment by wdigger** [comment]
I don't use Windows at all and MinGW too.
Try to use absolute path on configuring step (cmake .... e:\Dev\freeserf).

**Comment by ryan-de-boer** [comment]
Removed  /wd4290 from the 
freeserf\src\CMakeLists.txt and freeserf\tests\CMakeLists.txt
got further..

Had to also change savegame.cc eg KF_FLAG_CREATE does not exist.

Then there were lots of include paths that were not being supplied properly (just a workaround so it finds them for now).

There lots of linker errors - changed to dll libraries fixed those.

Now it creates FreeSerf.exe (fails on the tests though).
FreeSerf.exe with all the dependent dlls just runs and disappears. So close but does not work.

What windows platform works building HEAD?

**Comment by ryan-de-boer** [comment]
SOLVED yey
When FreeSerf.exe was disappearing with no errors, I modified the code to put logging into a file, debugged the problematic area, added SDL_GetError after init failed, and eventually it said "directx not available". I removed the env var $SDL_VIDEODRIVER=directx I had used to try and fix previous versions and then it ran fine!

So it does work in MinGW (with quite a few changes).

I'll close this issue for now. If I decide to get my changes committed I'll reopen it.
